pkg/api/validation/validation.go

```diff
@@ -2058,12 +2058,20 @@ func ValidateLimitRange(limitRange *api.LimitRange) field.ErrorList {
 		maxLimitRequestRatios := map[string]resource.Quantity{}
 
 		for k, q := range limit.Max {
-			allErrs = append(allErrs, validateContainerResourceName(string(k), idxPath.Child("max").Key(string(k)))...)
+			if limit.Type == api.LimitTypePod || limit.Type == api.LimitTypeContainer {
```

Comment:
can we just have a function like this so its not needing to be repeated everywhere in the if blocks?

```
func validateLimitRangeResourceName(limitType api.LimitType, resourceName string) {
 switch limitType {
   case api.LimitTypePod, api.LimitTypeContainer:
     return validateContainerResourceName()
   default:
     return validateResourceName()
 }
}
```